pandas/_libs/tslibs/fields.pyx

```diff
@@ -18,10 +18,15 @@ from pandas._libs.tslibs.ccalendar cimport (
     get_day_of_year)
 from pandas._libs.tslibs.np_datetime cimport (
     npy_datetimestruct, pandas_timedeltastruct, dt64_to_dtstruct,
-    td64_to_tdstruct)
+    td64_to_tdstruct, NPY_DATETIMEUNIT, NPY_FR_ns)
 from pandas._libs.tslibs.nattype cimport NPY_NAT
 
 
+cdef extern from "./src/datetime/np_datetime_strings.h":
+    int make_iso_8601_datetime(npy_datetimestruct *dts, char *outstr, int outlen,
+                               NPY_DATETIMEUNIT base)
```

Comment:
All else equal I'd rather do the "extern" in np_datetime and cimport from there.  Trying to minimize the surface area of "src".  That said, the ship may have sailed on that one, in which case never mind.